### PR TITLE
packaging/osbuilder: allow to pull and unpack pause image

### PR DESCRIPTION
**Pull request by ChengyuZhu6**
For Confidential containers stack, the pause image is managed by host side, then it may configure a malicious pause image, we need package a pause image inside the rootfs and don't the pause image from host. So the pause image will be part of the Kata Containers rootfs that's used by the Confidential Containers usecase.

Fixes: https://github.com/kata-containers/kata-containers/issues/9032

### CODE REVIEW

**Review by fidencio — COMMENTED** [review]
@ChengyuZhu6, I've left a few comments.
Thanks!

File: .github/workflows/build-kata-static-tarball-amd64.yaml [hunk 1]
```diff
@@ -44,6 +44,7 @@ jobs:
           - nydus
           - ovmf
           - ovmf-sev
+          - pause
```
[the remark this answers is not in the document]
Comment (ChengyuZhu6) [reply .github/workflows/build-kata-static-tarball-amd64.yaml]: Fixed.

File: .github/workflows/build-kata-static-tarball-s390x.yaml [hunk 2]
```diff
@@ -30,6 +30,7 @@ jobs:
           - agent
           - coco-guest-components
           - kernel
+          - pause
```
[the remark this answers is not in the document]
Comment (ChengyuZhu6) [reply .github/workflows/build-kata-static-tarball-s390x.yaml]: Fixed.

File: tools/osbuilder/rootfs-builder/rootfs.sh [hunk 3]
```diff
@@ -176,6 +177,12 @@ KERNEL_MODULES_DIR  Path to a directory containing kernel modules to include in
 LIBC                libc the agent is built against (gnu or musl).
                     Default value: ${LIBC} (varies with architecture)
 
+PAUSE_IMAGE_TARBALL Path to the kata-pause-image.tar.xz tarball to be unpacked inside the
```
[the remark this answers is not in the document]
Comment (ChengyuZhu6) [reply tools/osbuilder/rootfs-builder/rootfs.sh:180]: Fixed.

File: tools/packaging/static-build/pause-image/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+
+ENV DEBIAN_FRONTEND=noninteractive
+
+SHELL ["/bin/bash", "-o", "pipefail", "-c"]
+RUN apt-get update && \
+	apt-get --no-install-recommends -y install \
+    ca-certificates \
+    curl \
+    umoci \
+    skopeo
```
[the remark this answers is not in the document]
Comment (ChengyuZhu6) [reply tools/packaging/static-build/pause-image/Dockerfile:14]: Fixed.

File: tools/packaging/static-build/pause-image/build-static-pause-image.sh [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+source "${script_dir}/../../scripts/lib.sh"
+
+[ -n "$pause_image_repo" ] || die "failed to get pause-image repo"
+[ -n "$pause_image_version" ] || die "failed to get pause-image version"
+
+pull_pause_image_from_remote() {
+	echo "pull pause image from remote"
+
+	skopeo copy "${pause_image_repo}":"${pause_image_version}" oci:pause:"${pause_image_version}"
+	umoci unpack --image pause:"${pause_image_version}"  "${DESTDIR}/pause_bundle"
```
[the remark this answers is not in the document]
Comment (ChengyuZhu6) [reply tools/packaging/static-build/pause-image/build-static-pause-image.sh:22]: Very good point, @fidencio. I have removed `umoci.json` in `pause_bundle`. Thank you!

**Review by fidencio — APPROVED** [review]
lgtm, thanks @ChengyuZhu6!

**Review by stevenhorsman — APPROVED** [review]
One query here, but otherwise it looks good to me

File: tools/packaging/kata-deploy/local-build/kata-deploy-binaries.sh [hunk 6]
```diff
@@ -847,6 +849,23 @@ install_coco_guest_components() {
 	DESTDIR="${destdir}" "${coco_guest_components_builder}"
 }
 
+install_pause_image() {
+	latest_artefact="$(get_from_kata_deps "externals.pause.version")"
```
[the remark this answers is not in the document]
Comment (fidencio) [reply tools/packaging/kata-deploy/local-build/kata-deploy-binaries.sh:853]: Yeah, I agree. :-)
Thanks @stevenhorsman

[the remark this answers is not in the document]
Comment (ChengyuZhu6) [reply tools/packaging/kata-deploy/local-build/kata-deploy-binaries.sh]: Fixed.

**Comment by ChengyuZhu6** [comment]
@fidencio @stevenhorsman  I don’t think the failures are caused by the PR, but I’m not sure what the reason is. Do you have any suggestions? :disappointed_relieved:

**Comment by stevenhorsman** [comment]
> @fidencio @stevenhorsman I don’t think the failures are caused by the PR, but I’m not sure what the reason is. Do you have any suggestions? 😥

Unfortunately my first attempt is normally to re-run the failed tests, so I'll try that now and see if things improve. Leave it to us and don't worry about it yourself for now.

**Pull request merged**